src/main/kotlin/no/nav/syfo/service/SenderFacade.kt

```diff
@@ -20,26 +18,49 @@ import no.nav.syfo.kafka.producers.dittsykefravaer.DittSykefravaerMeldingKafkaPr
 import no.nav.syfo.kafka.producers.dittsykefravaer.domain.DittSykefravaerVarsel
 
 class SenderFacade(
-    val dineSykmeldteHendelseKafkaProducer: DineSykmeldteHendelseKafkaProducer,
-    val dittSykefravaerMeldingKafkaProducer: DittSykefravaerMeldingKafkaProducer,
-    val brukernotifikasjonerService: BrukernotifikasjonerService,
-    val arbeidsgiverNotifikasjonService: ArbeidsgiverNotifikasjonService,
-    val fysiskBrevUtsendingService: FysiskBrevUtsendingService,
-    val database: DatabaseInterface
+    private val dineSykmeldteHendelseKafkaProducer: DineSykmeldteHendelseKafkaProducer,
+    private val dittSykefravaerMeldingKafkaProducer: DittSykefravaerMeldingKafkaProducer,
+    private val brukernotifikasjonerService: BrukernotifikasjonerService,
+    private val arbeidsgiverNotifikasjonService: ArbeidsgiverNotifikasjonService,
+    private val fysiskBrevUtsendingService: FysiskBrevUtsendingService,
+    val database: DatabaseInterface,
 ) {
     fun sendTilDineSykmeldte(
         varselHendelse: NarmesteLederHendelse,
         varsel: DineSykmeldteVarsel,
     ) {
-        dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)
+        try {
+            dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)
+        } catch (e: Exception) {
+            log.error("Error while sending varsel to DINE_SYKMELDTE: ${e.message}", e)
+            lagreIkkeUtsendtNarmesteLederVarsel(
+                kanal = DINE_SYKMELDTE,
+                varselHendelse = varselHendelse,
+                eksternReferanse = varsel.id.toString(),
+                feilmelding = e.message,
+            )
+            throw Exception("Error while sending varsel to DINE_SYKMELDTE: ${e.message}", e)
+        }
```

Comment:
Som regel sendes varslene til flere kanaler av gangen, feks Dine sykmeldte og Arbeidsgivernotifikasjoner. Når vi kaster exception videre så er jeg redd det kan føre til at vi ikke sender til de andre kanalene. Hvis man feks skal sende til Dine sykmeldte og Arbeidsgivernotifikasjoner, og den første (til Dine sykmeldte) feiler, så vil vi ikke forsøke å sende til Arbeidsgivernotifikasjoner. Jeg synes vi skal bare logge og lagre i tabellen, men ikke kaste exception videre.
```suggestion
        try {
            dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)
            lagreUtsendtNarmesteLederVarsel(DINE_SYKMELDTE, varselHendelse, varsel.id.toString())
        } catch (e: Exception) {
            log.error("Error while sending varsel to DINE_SYKMELDTE: ${e.message}", e)
            lagreIkkeUtsendtNarmesteLederVarsel(
                kanal = DINE_SYKMELDTE,
                varselHendelse = varselHendelse,
                eksternReferanse = varsel.id.toString(),
                feilmelding = e.message,
            )
        }
```

Reply:
Det samme gjelder de andre kanalene

Reply:
Enig i det du sier.